File: backend/app/schemas/notification.py
import uuid
from datetime import datetime

from pydantic import BaseModel

from app.models.notification import NotificationPriority, NotificationType


class NotificationCreate(BaseModel):
    notification_type: NotificationType
    priority: NotificationPriority = NotificationPriority.NORMAL
    title: str
    message: str
    entity_type: str | None = None
    entity_id: uuid.UUID | None = None
    user_id: uuid.UUID | None = None  # None = broadcast


class NotificationResponse(BaseModel):
    id: uuid.UUID
    notification_type: NotificationType
    priority: NotificationPriority
    title: str
    message: str
    entity_type: str | None = None
    entity_id: uuid.UUID | None = None
    is_read: bool
    read_at: datetime | None = None
    user_id: uuid.UUID | None = None
    created_at: datetime

    model_config = {"from_attributes": True}


class NotificationListResponse(BaseModel):
    items: list[NotificationResponse]
    total: int
    unread_count: int


class UnreadCountResponse(BaseModel):
    count: int

File: frontend/src/types/notification.ts
export type NotificationType =
  | 'loyer_retard'
  | 'bail_expire_soon'
  | 'bail_expire'
  | 'paiement_recu'
  | 'systeme'

export type NotificationPriority = 'low' | 'normal' | 'high' | 'urgent'

export const NOTIFICATION_TYPE_LABELS: Record<NotificationType, string> = {
  loyer_retard: 'Loyer en retard',
  bail_expire_soon: 'Bail expirant bientôt',
  bail_expire: 'Bail expiré',
  paiement_recu: 'Paiement reçu',
  systeme: 'Système',
}

export const NOTIFICATION_PRIORITY_VARIANTS: Record<
  NotificationPriority,
  'red' | 'yellow' | 'blue' | 'gray'
> = {
  urgent: 'red',
  high: 'red',
  normal: 'blue',
  low: 'gray',
}

export interface Notification {
  id: string
  notification_type: NotificationType
  priority: NotificationPriority
  title: string
  message: string
  entity_type?: string
  entity_id?: string
  is_read: boolean
  read_at?: string
  user_id?: string
  created_at: string
}

export interface NotificationListResponse {
  items: Notification[]
  total: number
  unread_count: number
}

export interface UnreadCountResponse {
  count: number
}
